network/arista/eos_facts.py

```diff
@@ -0,0 +1,430 @@
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL ARISTA NETWORKS
+# BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+# OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
+# IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+DOCUMENTATION = """
+---
+module: eos_facts
+short_description: Gathers facts from Arista EOS nodes
+description:
+  - The eos_facts module collects facts from the EOS for use in
+    Ansible playbooks.  It can be used independently as well to discover
+    what facts are availble from the node.  This facts module does not
+    cache any facts.  If no configuration options are specified, then all
+    facts are returned.
+version_added: 1.0.0
+category: System
+author: Arista EOS+
+requirements:
+  - Arista EOS 4.13.7M or later with command API enabled
+  - Python Client for eAPI 0.3.0 or later
+notes:
+  - Supports eos metaparameters for using the eAPI transport
+  - The include and exclude options are mutually exclusive
+options:
+  include:
+    description:
+      - Specifies the list of facts to include when the fact module runs.  The
+        include list is comma delimited and, when included, will only
+        return the facts named in the include list.  All other facts will
+        not be returned.
+    required: false
+    default: null
+    choices: []
+    aliases: []
+    version_added: 1.0.0
+  exclude:
+    description:
+      - Specifies the list of facts to exclude when the fact module runs.  The
+        exclude list is comma delimited and, when configured, will not
+        return the facts named in the exclude list.  All other facts will
+        be returned.
+    required: false
+    default: null
+    choices: []
+    aliases: []
+    version_added: 1.0.0
+"""
+
+EXAMPLES = """
+
+- name: collect all facts from node
+  eos_facts:
+
+- name: include only a filtered set of facts returned
+  eos_facts: include=interfaces
+
+- name: exclude a specific set of facts
+  eos_facts: exclude=vlans
+
+"""
+#<<EOS_COMMON_MODULE_START>>
+
+import syslog
+import collections
+
+from ansible.module_utils.basic import *
+
+try:
+    import pyeapi
+    PYEAPI_AVAILABLE = True
+except ImportError:
+    PYEAPI_AVAILABLE = False
+
+DEFAULT_SYSLOG_PRIORITY = syslog.LOG_NOTICE
+DEFAULT_CONNECTION = 'localhost'
+TRANSPORTS = ['socket', 'http', 'https', 'http_local']
+
+class EosAnsibleModule(AnsibleModule):
```

Comment:
this class should be moved to common file under module_utils instead of repeating it in each module